### ISSUE TITLE
C++ debugging on macOS gets stuck with message "Unrecognized format of field "msg" in result"

### ISSUE THREAD

**Issue by kaloianm** [microsoft/vscode-cpptools#2492]
Hello,

I am occasionally using VSCode and the Microsoft C++ extension for MongoDB server development on macOS. When I try to debug any of the produced binaries (which are compiled with debugging information enabled and no optimizations), the debugger just gets stuck and the process remains stuck in unkillable state.

OS version: macOS 10.13.6
VS Code Version: 1.27.1
C/C++ Extension Version: 0.18.1

My debugger configuration looks like this:
```
        {
            "name": "(lldb) Generic run",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceFolder}/build/ninja/mongo/db/s/collection_sharding_runtime_test",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": true,
            "MIMode": "lldb"
        }
```

When I start this debugger configuration with breakpoint enabled, I am able to step through the code successfully, up to a point when an exception gets thrown at which case the debugger gets stuck and the following messages is printed on the console:
```
...
Loaded '/Users/kaloianm/workspace/mongo/build/ninja/mongo/db/s/collection_sharding_runtime_test'. Symbols loaded.
Execute debugger commands using "-exec <command>", for example "-exec info registers" will list registers in use (when GDB is the debugger)
ERROR: Exception while processing MIEngine operation. Unrecognized format of field "msg" in result: result-class: error
message: error: Failed to halt process: Process is not running. If the problem continues restart debugging.
```

What I noticed is that from the 4 threads that are in the process, one remains in the PAUSED mode and I can inspect its stack. The other 3 are unpausable and unkillable.

Let me know if I can provide any additional information to help debug this.

-Kal.

**Comment by pieandcakes** [comment]
@kaloianm Can you enable [engine logging](https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/FAQs.md#how-to-enable-logging) and see what the error in the msg field is?


>What I noticed is that from the 4 threads that are in the process, one remains in the PAUSED mode and I can inspect its stack. The other 3 are unpausable and unkillable.

Is this from the UI? Can you confirm by trying `-exec thread list` in the VS Code debug console to see
if they are actually stopped? Our debug engine should always stop all threads and continue all threads so that is sort of strange.

**Comment by modavi** [comment]
I've been running into the same issue using watchpoints. I've created a simple c app that increments a local variable, then I break and set a watchpoint in the debug console window with "-exec wa s v test" and then as soon as my variable is modified, the IDE appears to exit the debug state and appears as though the app is running (although it's not). 

When i go to pause the app again I get the above error as well. Doing this in a larger multithreaded app, the UI does indicate one thread is 'running' while the other 3 are paused, though I can't run the '-exec thread list' command as the vscode debugger interface does not appear to be in the paused state even though lldb is.

I've attached the output from running with engine logging.

[engine-log.txt](https://github.com/microsoft/vscode-cpptools/files/3999138/engine-log.txt)

**Comment by pieandcakes** [comment]
@modavi After looking at the log, this is what I see from what you are saying:
```
1: (9283) <-1025-exec-next --thread 1 --frame 0
1: (9283) ->1025^running
1: (9284) 1025: elapsed time 1
1: (9284) ->(gdb)
1: (9285) ->(gdb)
1: (9285) ->*running,thread-id="all"
1: (9285) ->(gdb)
1: (9285) ->(gdb)
1: (16948) <-1026-exec-interrupt
1: (16948) ->1026^error,message="error: Failed to halt process: Process is not running."
```

We are telling the debugger to step using `-exec-next` and we receive the response `^running`. The problem is that `lldb` is not telling us it has stopped from that state, which is why we haven't stopped. 

Earlier in the log there is this block:
```
1: (5211) <-1017-interpreter-exec console "wa s v test"
1: (5212) ->Watchpoint created: Watchpoint 1: addr = 0x7ffeefbffe08 size = 4 state = enabled type = w
1: (5212) ->    declare @ '/Users/morgan/Desktop/TestCProject/main.c:4'
1: (5212) ->    watchpoint spec = 'test'
1: (5212) ->    new value: 0
1: (5212) ->1017^done
1: (5212) ->(gdb)
1: (5212) 1017: elapsed time 1
Watchpoint created: Watchpoint 1: addr = 0x7ffeefbffe08 size = 4 state = enabled type = w
    declare @ '/Users/morgan/Desktop/TestCProject/main.c:4'
    watchpoint spec = 'test'
    new value: 0

1: (7367) <-1018-exec-next --thread 1 --frame 0
1: (7368) ->1018^running
1: (7368) 1018: elapsed time 1
1: (7368) ->(gdb)
1: (7369) ->(gdb)
1: (7369) ->*running,thread-id="all"
1: (7369) ->(gdb)
1: (7372) ->(gdb)
1: (7372) ->*stopped,reason="end-stepping-range",frame={level="0",addr="0x0000000100000f93",func="main",args=[],file="main.c",fullname="/Users/morgan/Desktop/TestCProject/main.c",line="8"},thread-id="1",stopped-threads="all"
1: (7372) ->(gdb)
```
Which lldb notifies us that it has stopped because it has hit its stepping range using `*stopped,reason="end-stepping-range"` which is what we're expecting with the second call but not receiving. As to why, I don't know at this time and would need to see where the bug is in `lldb`. Can you share your `main.c`?


The error about the `msg` field looks like a bug. I'll see if I can fix that but even if I fix it, the underlying issue is still there and debugging will stop.

**Comment by modavi** [comment]
I don't believe I have the original main.c anymore, but it was literally just a sample like this...

```
int main() {
   int test = 0;
   test += 1;   
   return 0;
}
```

**Comment by sean-mcmanus** [comment]
Fixed in our latest Insiders release: https://github.com/microsoft/vscode-cpptools/releases/tag/0.27.0-insiders. But there's a known issue
```
Pause (break all) doesn't work when debugging on Mac. Workaround: use kill -2 <pid> to trigger the break.
```
